src/Socket.cpp

```diff
@@ -29,6 +30,13 @@ Socket::Socket() : d(new Private)
 
 Socket::~Socket()
 {
+    {
+        std::lock_guard<std::mutex> lk(d->receiveQueueMutexBlock);
+        d->message_ready = true;
+    }
+
+    d->socket_block_condition_variable.notify_all();
```

Comment:
Shouldn't this be `notify_one`?

Reply:
I am not sure, the difference that it will notify all waiting threads. So far as I understand we have only one thread, but I might be wrong.